### PR TITLE
stt: use vector in flute

### PR DESCRIPTION
**Pull request by eder-matheus**

### CODE REVIEW

File: src/grt/test/pd3.tcl [hunk 1]
```diff
@@ -5,7 +5,7 @@ read_def "gcd.def"
 
 set guide_file [make_result_file pd3.guide]
 
-set_routing_alpha 0.9 -min_fanout 10
+set_routing_alpha 0.9 -min_fanout 9
```
Comment (maliberty) [inline src/grt/test/pd3.tcl:8]: Why is this changed?

File: src/grt/src/fastroute/include/FastRoute.h [hunk 2]
```diff
@@ -249,16 +249,14 @@ class FastRouteCore
                     bool newType,
                     bool noADJ);
   void fluteNormal(int netID,
-                   int d,
-                   DTYPE x[],
-                   DTYPE y[],
+                   std::vector<DTYPE> x,
+                   std::vector<DTYPE> y,
```
Comment (maliberty) [inline src/grt/src/fastroute/include/FastRoute.h:253]: Use a reference type to avoid copying, preferably const&

File: src/grt/src/fastroute/include/FastRoute.h [hunk 3]
```diff
@@ -249,16 +249,14 @@ class FastRouteCore
...
                    int acc,
                    float coeffV,
                    Tree* t);
   void fluteCongest(int netID,
-                    int d,
-                    DTYPE x[],
-                    DTYPE y[],
+                    std::vector<DTYPE> x,
+                    std::vector<DTYPE> y,
```
Comment (maliberty) [inline src/grt/src/fastroute/include/FastRoute.h:259]: Use a reference type to avoid copying, preferably const&

File: src/grt/src/fastroute/src/RSMT.cpp [hunk 4]
```diff
@@ -61,7 +61,7 @@ static int ordery(const struct pnt* a, const struct pnt* b)
 }
 
 // binary search to map the new coordinates to original coordinates
-int mapxy(int nx, int xs[], int nxs[], int d)
+int mapxy(int nx, std::vector<int> xs, std::vector<int> nxs, int d)
```
Comment (maliberty) [inline src/grt/src/fastroute/src/RSMT.cpp:64]: Use a reference type to avoid copying, preferably const&

File: src/grt/src/fastroute/src/RSMT.cpp [hunk 5]
```diff
@@ -154,19 +154,20 @@ void FastRouteCore::copyStTree(int ind, Tree rsmt)
 }
 
 void FastRouteCore::fluteNormal(int netID,
-                 int d,
-                 DTYPE x[],
-                 DTYPE y[],
+                 std::vector<DTYPE> x,
+                 std::vector<DTYPE> y,
```
Comment (maliberty) [inline src/grt/src/fastroute/src/RSMT.cpp:158]: Use a reference type to avoid copying, preferably const&

File: src/stt/include/stt/SteinerTreeBuilder.h [hunk 6]
```diff
@@ -73,10 +73,9 @@ class SteinerTreeBuilder
                        int drvr_index);
   // API only for FastRoute, that requires the use of flutes in its
   // internal flute implementation
-  Tree makeSteinerTree(int num_pins,
-                       int xs[],
-                       int ys[],
-                       int s[],
+  Tree makeSteinerTree(std::vector<int>& x,
+                       std::vector<int>& y,
+                       std::vector<int>& s,
```
Comment (maliberty) [inline src/stt/include/stt/SteinerTreeBuilder.h:78]: const& ?

File: src/stt/include/stt/flute.h [hunk 7]
```diff
@@ -72,58 +72,59 @@ using stt::Tree;
...
-inline DTYPE flutes_wl_LMD(int d, DTYPE xs[], DTYPE ys[], int s[], int acc) {
+DTYPE flutes_wl_LD(int d, std::vector<DTYPE> xs, std::vector<DTYPE> ys, std::vector<int> s);
+DTYPE flutes_wl_MD(int d, std::vector<DTYPE> xs, std::vector<DTYPE> ys, std::vector<int> s, int acc);
+DTYPE flutes_wl_RDP(int d, std::vector<DTYPE> xs, std::vector<DTYPE> ys, std::vector<int> s, int acc);
+Tree flutes_LD(int d, std::vector<DTYPE> xs, std::vector<DTYPE> ys, std::vector<int> s);
+Tree flutes_MD(int d, std::vector<DTYPE> xs, std::vector<DTYPE> ys, std::vector<int> s, int acc);
+Tree flutes_HD(int d, std::vector<DTYPE> xs, std::vector<DTYPE> ys, std::vector<int> s, int acc);
+Tree flutes_RDP(int d, std::vector<DTYPE> xs, std::vector<DTYPE> ys, std::vector<int> s, int acc);
+
+inline DTYPE flutes_wl_LMD(int d, std::vector<DTYPE> xs, std::vector<DTYPE> ys, std::vector<int> s, int acc) {
```
Comment (maliberty) [inline src/stt/include/stt/flute.h:90]: Use a reference type to avoid copying, preferably const&

File: src/stt/src/SteinerTreeBuilder.cpp [hunk 8]
```diff
@@ -90,13 +90,12 @@ Tree SteinerTreeBuilder::makeSteinerTree(odb::dbNet* net,
   return tree;
 }
 
-Tree SteinerTreeBuilder::makeSteinerTree(int num_pins,
-                                         int x[],
-                                         int y[],
-                                         int s[],
+Tree SteinerTreeBuilder::makeSteinerTree(std::vector<int>& x,
+                                         std::vector<int>& y,
+                                         std::vector<int>& s,
```
Comment (maliberty) [inline src/stt/src/SteinerTreeBuilder.cpp:95]: Use a reference type to avoid copying, preferably const&

**Review by maliberty — CHANGES_REQUESTED** [review]

File: src/grt/test/pd3.tcl (hunk 1 above)
Comment (eder-matheus) [reply src/grt/test/pd3.tcl:8]: I've updated the check to be consistent with the option name (fanout = term_count -1). Too keep the ok files for this test, I've just updated the script.

**Review by maliberty — APPROVED** [review]
